enterprise/metrics/src/main/java/org/neo4j/metrics/output/CsvReporterSingle.java

```diff
@@ -0,0 +1,379 @@
+        {
+            this.clock = clock;
+            return this;
+        }
+
+        /**
+         * Only report metrics which match the given filter.
+         *
+         * @param filter a {@link MetricFilter}
+         * @return {@code this}
+         */
+        public Builder filter( MetricFilter filter )
+        {
+            this.filter = filter;
+            return this;
+        }
+
+        /**
+         * Builds a {@link CsvReporterSingle} with the given properties, writing {@code .csv} data to the
+         * given file.
+         *
+         * @param file the file in which the {@code .csv} data will be inserted
+         * @return a {@link CsvReporterSingle}
+         */
+        public CsvReporterSingle build( File file )
+        {
+            return new CsvReporterSingle( registry,
+                    file,
+                    rateUnit,
+                    durationUnit,
+                    clock,
+                    filter );
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger( CsvReporter.class );
+    private static final Charset UTF_8 = Charset.forName( "UTF-8" );
+    private static final SimpleDateFormat ISO8601 = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss" );
+
+    private final File file;
+    private final Clock clock;
+    private PrintWriter out;
+
+    private CsvReporterSingle( MetricRegistry registry,
+            File file,
+            TimeUnit rateUnit,
+            TimeUnit durationUnit,
+            Clock clock,
+            MetricFilter filter )
+    {
+        super( registry, "csv-reporter-single", filter, rateUnit, durationUnit );
+        this.file = file;
+        this.clock = clock;
+    }
+
+    @Override
+    public void stop()
+    {
+        super.stop();
+
+        if ( out != null )
+        {
+            out.close();
+        }
+    }
+
+    @Override
+    public void report( SortedMap<String,Gauge> gauges,
+            SortedMap<String,Counter> counters,
+            SortedMap<String,Histogram> histograms,
+            SortedMap<String,Meter> meters,
+            SortedMap<String,Timer> timers )
+    {
+        if ( out == null )
+        {
+            try
+            {
+                if ( file.exists() )
+                {
+                    // Archive existing log
+                    String fileName = file.getName();
+                    fileName = new SimpleDateFormat( "yyyy_MM_dd_HH_mm_ss" ).format( new Date( file.lastModified() ) ) +
+                               fileName;
+                    File archiveFile = new File( file.getParentFile(), fileName );
+                    if ( !file.renameTo( archiveFile ) )
+                    {
+                        throw new IllegalStateException( "Could not move old metrics log to " + archiveFile );
+                    }
+                }
+
+                out = new PrintWriter( new OutputStreamWriter( new FileOutputStream( file ), UTF_8 ) );
```

Comment:
Since we already have file and encoding we can use `PrintWriter(String fileName, String csn)` in that case we can avoid stream creation